src/app/qgsnewspatialitelayerdialog.cpp

```diff
@@ -55,18 +55,20 @@ QgsNewSpatialiteLayerDialog::QgsNewSpatialiteLayerDialog( QWidget *parent, Qt::W
   QgsSettings settings;
   restoreGeometry( settings.value( QStringLiteral( "Windows/NewSpatiaLiteLayer/geometry" ) ).toByteArray() );
 
-  mGeometryTypeBox->addItem( tr( "Point" ), QStringLiteral( "POINT" ) );
-  mGeometryTypeBox->addItem( tr( "Line" ), QStringLiteral( "LINE" ) );
-  mGeometryTypeBox->addItem( tr( "Polygon" ), QStringLiteral( "POLYGON" ) );
-  mGeometryTypeBox->addItem( tr( "MultiPoint" ), QStringLiteral( "MULTIPOINT" ) );
-  mGeometryTypeBox->addItem( tr( "MultiLine" ), QStringLiteral( "MULTILINE" ) );
-  mGeometryTypeBox->addItem( tr( "MultiPolygon" ), QStringLiteral( "MULTIPOLYGON" ) );
+  QString path = QgsApplication::defaultThemePath();
+
+  mGeometryTypeBox->addItem( QIcon( path + "/mIconPointLayer.svg" ), tr( "Point" ), QStringLiteral( "POINT" ) );
```

Comment:
One more thing - instead of manually building the icon paths, you should be using QgsApplication::getThemeIcon( QStringLiteral("mIconPointLayer.svg")) instead. It's safer, plus much faster since the icons are cached.